src/jit/gentree.h

```diff
@@ -574,11 +574,21 @@ struct GenTree
         return isContained() && IsCnsIntOrI() && !isUsedFromSpillTemp();
     }
 
+#if FEATURE_X87_DOUBLES
```

Comment:
I think if we are actually violating 2, then it needs to be fixed as it would mean we aren't following the IEEE 754 spec, and by extension the ECMA-335 spec (which states we follow the IEEE 754 spec for all floating point numbers).

If we aren't violating 2, then this change won't break anything and is an optimization and could potentially make future optimizations easier. We would only be taking up 4 less bytes per float constant we stored and would not have to upcast/downcast anytime we had to deal with the value.